### PR TITLE
Fiterrs bug in MCMCfitter and Event Optimize

### PR DESCRIPTION
**Pull request by devbhakt**
A fix for issue #1067

### CODE REVIEW

File: src/pint/scripts/event_optimize.py [hunk 1]
```diff
@@ -241,15 +241,18 @@ def get_fit_keyvals(model, phs=0.0, phserr=0.1):
     fitkeys = [p for p in model.params if not getattr(model, p).frozen]
     fitvals = []
     fiterrs = []
-    for p in fitkeys:
+    fitunits = []
+    for i, p in enumerate(fitkeys):
         fitvals.append(getattr(model, p).value)
-        fiterrs.append(getattr(model, p).uncertainty_value)
+        fitunits.append(getattr(model, p).units)
+        fiterrs.append(float(((getattr(model, p).uncertainty).to(fitunits[i])).value))
```
Comment (aarchiba) [inline src/pint/scripts/event_optimize.py:248]: Do you actually need the list of fitunits? It seems like the units are available from `model.PARAM.uncertainty.unit`?

File: src/pint/scripts/event_optimize.py [hunk 2]
```diff
@@ -241,15 +241,18 @@ def get_fit_keyvals(model, phs=0.0, phserr=0.1):
...
     # The last entry in each of the fit lists is our absolute PHASE term
     # Hopefully this will become a full PINT model param soon.
     fitkeys.append("PHASE")
     fitvals.append(phs)
     fiterrs.append(phserr)
-    return fitkeys, np.asarray(fitvals), np.asarray(fiterrs)
+    fitunits.append("pulse phase")
```
Comment (aarchiba) [inline src/pint/scripts/event_optimize.py:254]: Does the string `"pulse phase"` work as an acceptable unit?

File: src/pint/scripts/event_optimize.py (hunk 1 above)
Comment (devbhakt) [reply src/pint/scripts/event_optimize.py:248]: I was originally using fitunits for axis labels within the plotting priors method. I found a work around to making a list of fitunits so I can remove that list.

I tried both `model.PARAM.uncertainty.unit` and `model.params.uncertainty` and received the errors: Attribute PARAM not found in TimingModel or any Component and 'list' object has no attribute 'uncertainty'. I am not sure of those are bugs or I made a mistake in loading in the model.

File: src/pint/scripts/event_optimize.py (hunk 2 above)
Comment (devbhakt) [reply src/pint/scripts/event_optimize.py:254]: I had originally used that as a place holder and am working on changing that as it is not an acceptable unit. Thank you for pointing that out!

File: src/pint/scripts/event_optimize.py (hunk 1 above)
Comment (aarchiba) [reply src/pint/scripts/event_optimize.py]: Oh, uh, I was taking a shortcut. What I meant was, if your parameter was RAJ, `model.RAJ.uncertainty.unit` or `model.RAJ.units`; these are the same thing you'd get with `getattr(model, "RAJ").uncertainty.unit` or whatever.

**Comment by aarchiba** [comment]
I'm sorry we have to keep manually running your CI tasks - after we merge this PR it should be automatic, but until then each change has to be approved: https://github.blog/2021-04-22-github-actions-update-helping-maintainers-combat-bad-actors/

This is annoying - surely after the first time I approve your CI runs it could keep running them? - but in the meantime `tox` should do a fairly thorough job of running CI jobs locally. (Just run `tox` and it should run a bunch of versions with different dependencies; there's a command-line option to parallelize.)

**Comment by aarchiba** [comment]
Can you write a test case that would have failed before this change but passes now? As simple and fast as possible - maybe just one MCMC iteration, we don't need actual statistics.

**Comment by devbhakt** [comment]
I think this pull request can be closed without merging. I was writing the test case when I found that the original way, that event optimize and the mcmc fitter were getting the uncertainties, was providing the uncertainty values with the right corresponding units. I think the original issue was corrected by the Parameter simplification pull request #1071

**Comment by aarchiba** [comment]
Would you mind putting the test case in the PR? It was a bug once and might be again if we aren't careful. For that matter if someone accidentally fixed the code, someone (ahem) might accidentally break it again.

**Comment by devbhakt** [comment]
Pull Request #1071 has the test_units_consistent testcase within tests_parameters.py which covers the check for the model parameter units and model error units.
I don't think we need to do a test case with a MCMC iteration because the MCMC fitter will run to completion before and after the fix.

**Pull request closed without merging**